modules/tensor_mechanics/src/materials/RankTwoInvariant.C

```diff
@@ -60,31 +61,43 @@ RankTwoInvariantTempl<is_ad>::computeQpProperties()
 {
   switch (_invariant)
   {
-    case RankTwoScalarTools::INVARIANT_TYPE::MaxPrincipal:
-    case RankTwoScalarTools::INVARIANT_TYPE::MidPrincipal:
-    case RankTwoScalarTools::INVARIANT_TYPE::MinPrincipal:
+    case RankTwoScalarTools::InvariantType::MaxPrincipal:
+    case RankTwoScalarTools::InvariantType::MidPrincipal:
+    case RankTwoScalarTools::InvariantType::MinPrincipal:
     {
       Point dummy_direction;
       _property[_qp] = RankTwoScalarTools::getPrincipalComponent(
           MetaPhysicL::raw_value(_tensor[_qp]), _invariant, dummy_direction);
       break;
     }
-    case RankTwoScalarTools::INVARIANT_TYPE::VonMisesStress:
-    case RankTwoScalarTools::INVARIANT_TYPE::EffectiveStrain:
-    case RankTwoScalarTools::INVARIANT_TYPE::Hydrostatic:
-    case RankTwoScalarTools::INVARIANT_TYPE::L2norm:
-    case RankTwoScalarTools::INVARIANT_TYPE::VolumetricStrain:
-    case RankTwoScalarTools::INVARIANT_TYPE::FirstInvariant:
-    case RankTwoScalarTools::INVARIANT_TYPE::SecondInvariant:
-    case RankTwoScalarTools::INVARIANT_TYPE::ThirdInvariant:
-    case RankTwoScalarTools::INVARIANT_TYPE::TriaxialityStress:
-    case RankTwoScalarTools::INVARIANT_TYPE::MaxShear:
-    case RankTwoScalarTools::INVARIANT_TYPE::StressIntensity:
+
+    case RankTwoScalarTools::InvariantType::VonMisesStress:
+    case RankTwoScalarTools::InvariantType::Hydrostatic:
+    case RankTwoScalarTools::InvariantType::L2norm:
+    case RankTwoScalarTools::InvariantType::VolumetricStrain:
+    case RankTwoScalarTools::InvariantType::FirstInvariant:
+    case RankTwoScalarTools::InvariantType::SecondInvariant:
+    case RankTwoScalarTools::InvariantType::ThirdInvariant:
+    case RankTwoScalarTools::InvariantType::TriaxialityStress:
+    case RankTwoScalarTools::InvariantType::MaxShear:
+    case RankTwoScalarTools::InvariantType::StressIntensity:
     {
       _property[_qp] =
           RankTwoScalarTools::getInvariant(MetaPhysicL::raw_value(_tensor[_qp]), _invariant);
       break;
     }
+
+    case RankTwoScalarTools::InvariantType::EffectiveStrain:
+    {
+      mooseAssert(_tensor_old, "The selected invariant requires the input material to be stateful");
+      mooseAssert(_property_old,
+                  "The selected invariant requires the output material to be stateful");
+      _property[_qp] = (*_property_old)[_qp] + RankTwoScalarTools::effectiveStrain(
```

Comment:
Do you think it's worth renaming `RankTwoScalarTools::effectiveStrain()` to `RankTwoScalarTools::effectiveStrainIncrement()` and having it take two arguments (the new and old tensors)? That would help reinforce the point that we can't directly compute that quantity from just the current state of the tensor.

Along the same lines, if we do that we should probably rename the enum entry for that from `RankTwoScalarTools::InvariantType::EffectiveStrain` to `RankTwoScalarTools::InvariantType::EffectiveStrainIncrement`.

Reply:
There is no API for processing a pair of tensors in teh corresponding function that takes that enum.